egs/formosa/s5/run.sh

```diff
@@ -0,0 +1,435 @@
+  exp/tri2/graph data/test exp/tri2/decode_test
+)&
+
+
+echo
+echo "========== tri2 done =========="
+echo
+
+#
+# tri3a
+#
+
+echo
+echo "========== tri3 =========="
+echo
+
+true && (
+
+# Train tri3a, which is LDA+MLLT,
+steps/train_lda_mllt.sh --cmd "$train_cmd" \
+ 2500 20000 data/train_tri3 data/lang exp/tri2_ali_tri3 exp/tri3a || exit 1;
+
+)
+
+# decode tri3a
+true && (
+utils/mkgraph.sh data/lang_test exp/tri3a exp/tri3a/graph || exit 1;
+steps/decode.sh --cmd "$decode_cmd" --nj $numjobs_test --config conf/decode.config \
+  exp/tri3a/graph data/test exp/tri3a/decode_test
+)&
+
+echo
+echo "========== tri3 done =========="
+echo
+
+#
+# tri4
+#
+
+echo
+echo "========== tri4 =========="
+echo
+
+true && (
+
+# From now, we start building a more serious system (with SAT), and we'll
+# do the alignment with fMLLR.
+steps/align_fmllr.sh --cmd "$train_cmd" --nj $numjobs \
+  data/train_tri4 data/lang exp/tri3a exp/tri3a_ali_tri4 || exit 1;
+
+steps/train_sat.sh --cmd "$train_cmd" \
+  2500 20000 data/train_tri4 data/lang exp/tri3a_ali_tri4 exp/tri4a || exit 1;
+
+# align tri4a
+steps/align_fmllr.sh  --cmd "$train_cmd" --nj $numjobs \
+  data/train data/lang exp/tri4a exp/tri4a_ali
+
+)
+
+# decode tri4a
+true && (
+utils/mkgraph.sh data/lang_test exp/tri4a exp/tri4a/graph
+steps/decode_fmllr.sh --cmd "$decode_cmd" --nj $numjobs_test --config conf/decode.config \
+  exp/tri4a/graph data/test exp/tri4a/decode_test
+)&
+
+echo
+echo "========== tri4 done =========="
+echo
+
+#
+# tri5
+#
+
+echo
+echo "========== tri5 =========="
+echo
+
+true && (
+
+# Building a larger SAT system.
+steps/train_sat.sh --cmd "$train_cmd" \
+  3500 100000 data/train data/lang exp/tri4a_ali exp/tri5a || exit 1;
+
+# align tri5a
+steps/align_fmllr.sh --cmd "$train_cmd" --nj $numjobs \
+  data/train data/lang exp/tri5a exp/tri5a_ali || exit 1;
+
+)
+
+# decode tri5
+true && (
+utils/mkgraph.sh data/lang_test exp/tri5a exp/tri5a/graph || exit 1;
+steps/decode_fmllr.sh --cmd "$decode_cmd" --nj $numjobs_test --config conf/decode.config \
+   exp/tri5a/graph data/test exp/tri5a/decode_test || exit 1;
+)&
+
+echo
+echo "========== tri5 done =========="
+echo
+
+#
+# nnet3 tdnn models
+#
+
+echo
+echo "========== nnet3 =========="
+echo
+
+true && (
+
+# option to bypass certain steps, for example
+# local/nnet3/run_tdnn.sh --stage 8 --train_stage 0	--> bypass feature extreaction 
+# local/nnet3/run_tdnn.sh --stage 9			--> decoding only
+#
+local/nnet3/run_tdnn.sh
+
+)
+
+echo
+echo "========== nnet3 done =========="
+echo
+
+#
+# chain model
+#
+
+echo
+echo "========== chain =========="
+echo
+
+true && (
+
+# There are options to bypass certain steps, for example
+# local/chain/run_tdnn.sh --stage 8			--> bypass feature extraction, if you have already run "local/nnet3/run_tdnn.sh"
+# local/chain/run_tdnn.sh --stage 11 --train_stage 120	--> continue the training from iteration 120
+local/chain/run_tdnn.sh --stage 8
```

Comment:
you can mention the --stage options in comments, but have the default be one that works out of the box.
Please remove most of the echo statements; if you want, approximately once per stage, you can have a statement like:
   echo "$0: training LDA+MLLT system"

Reply:
Thanks! All empty echo statements were removed.  Now only one echo per stage left and changed its format to echo "$0: ......"